compiler/src/iree/compiler/Dialect/Flow/Transforms/CollapseDimensions.cpp

```diff
@@ -792,36 +804,43 @@ static bool collapseDimensionsForDispatch(IRRewriter &rewriter,
     llvm::dbgs() << "\n";
   });
 
-  // Step 4. For each producer, reduce the number of collapsed dimensions
-  // based on the dimensions that it's consumers can collapse.
-  updateProducersFromConsumers(slice.getArrayRef(), opMap);
-
-  LLVM_DEBUG({
-    llvm::dbgs() << "[CollapseDims] : After updating producers: \n";
-    for (auto &[op, info] : opMap) {
-      info.dump();
-      llvm::dbgs() << "\n";
-      op.dump();
+  bool didUpdateProducers = true;
+  bool didUpdateConsumers = true;
+  while (didUpdateProducers || didUpdateConsumers) {
```

Comment:
You might also want to add a cut-off of max-num iterations, and bail if you broke out of the loop on hitting max number of iterations (like 5 or 10 iterations is fine cause this is within a dispatch)